sdk/digitaltwins/Azure.DigitalTwins.Core/src/Serialization/OptionalETagConverter.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DigitalTwins.Core.Serialization
+{
+    // TODO: Remove when #16272 is fixed
+    internal class OptionalETagConverter : JsonConverter<ETag?>
+    {
+        public override ETag? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            string value = reader.GetString();
+            return value != null
+                ? new ETag(value)
+                : (ETag?)null;
+        }
+
+        public override void Write(Utf8JsonWriter writer, ETag? value, JsonSerializerOptions options)
+        {
+            if (value == null)
+            { writer.WriteNullValue(); }
```

Comment:
won't this NRE?

Reply:
If the `writer` is null?  That won't happen with STJ and it's not a publicly exposed type.  We can add an `Argument.NotNull(writer, nameof(writer));` if we think it's possible customers could somehow get here.

Reply:
Ah, I misread this as `if (write == null)` for some reason.